test/integration/utility.h

```diff
@@ -22,7 +23,10 @@ class BufferingStreamDecoder : public Http::StreamDecoder, public Http::StreamCa
   BufferingStreamDecoder(std::function<void()> on_complete_cb) : on_complete_cb_(on_complete_cb) {}
 
   bool complete() { return complete_; }
-  const Http::HeaderMap& headers() { return *headers_; }
+  const Http::HeaderMap& headers() {
+    ASSERT(headers_ != nullptr);
```

Comment:
Better would probably be to just make this return a pointer instead of a reference, but that's a larger change. I don't feel very strongly about this either way.